rules/no-for-loop.js

```diff
@@ -83,30 +79,18 @@ const getArrayIdentifierNameFromBinaryExpression = (binaryExpression, indexIdent
 		return;
 	}
 
-	if (greater.property.name !== 'length') {
-		return;
-	}
-
-	return greater.object.name;
+	return greater.property.name !== 'length' ? undefined : greater.object.name;
```

Comment:
It looks like this function could be merged into 2 `return` statements, but indeed the readability could potentially be trash, especially if there are comments explaining each step.